reconcile/test/test_utils_ldap_client.py

```diff
@@ -0,0 +1,50 @@
+import ldap3
+import pytest
+from reconcile.utils.ldap_client import LdapClient
+
+
+@pytest.fixture
+def connection_search_result():
+    result = [
+        {"attributes": {"uid": ["user1"]}},
+        {"attributes": {"uid": ["user2"]}},
+        {"attributes": {"uid": ["user3"]}},
+    ]
+    return result
+
+
+def test_ldap_client_from_settings(mocker):
+    mock_connection_bind = mocker.patch(
+        "reconcile.utils.ldap_client.Connection", autospec=True
+    )
+
+    settings = {"ldap": {"baseDn": "test", "serverUrl": "testUrl"}}
+    with LdapClient.from_settings(settings) as ldap_client:
+        assert ldap_client.base_dn == "test"
+        assert mock_connection_bind.call_args.args[0].host == "testUrl"
+
+
+def test_ldap_client(mocker, connection_search_result):
+    mocked_connection = mocker.Mock(spec=ldap3.Connection)
```

Comment:
No patching, I like it!

I think you just want `mocker.create_autospec` here. Check out the docs in `unittest.mock`.